### PR TITLE
`Development`: Add templates for pull requests and issues

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**

### CODE REVIEW

File: .github/PULL_REQUEST_TEPLATE.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+<!-- Checklist for all PRs -->
+#### General
+- [ ] I chose a title conforming to the [naming conventions for pull requests](#).
+
+<!-- Checklist for PRs that affect our API-->
+#### AtlasML - API
+- [ ] I updated the API documentation.
+
+#### AtlasML - Machine Learning
+- [ ] I updated the machine learning documentation.
```
Comment (ardakaraman0) [inline .github/PULL_REQUEST_TEPLATE.md:30]: Do we have a documentation online? If so, we could put the link here for ease of use. Maybe we can note down small items (just one word) for the sections that were affected?

**Comment by ardakaraman0** [comment]
Also as an open thread, would we be interested in a template for the Pull Request where on the down bottom there can be a section where 
"As the reviewer I have, [ ] checked the ticket description, [ ] checked whether the code is written following the rules, [ ] tested on [ ] Google Chrome, [ ] Safari, etc, ....." 
and bunch of check items (tick boxes) for the reviewer to check off?

File: .github/PULL_REQUEST_TEPLATE.md (hunk 1 above)
Comment (ufukygmr) [reply .github/PULL_REQUEST_TEPLATE.md:30]: We can use Pydoc for documentation. Where we need to write Markdown for documentation. So, in this case we don't need a link or something like that. Everything will be readble in PRs

**Comment by ufukygmr** [comment]
> Also as an open thread, would we be interested in a template for the Pull Request where on the down bottom there can be a section where "As the reviewer I have, [ ] checked the ticket description, [ ] checked whether the code is written following the rules, [ ] tested on [ ] Google Chrome, [ ] Safari, etc, ....." and bunch of check items (tick boxes) for the reviewer to check off?

I would like to avoid at all. Since we will be writing a microservice, what they mostly do will be go over the code andrunning the microservice - testing the endpoints etc. So, I don't think we will intense review processes for a while. WE can skip this part for now. I think this will be needed when Atlas migration started.

**Review by ufukygmr — APPROVED** [review]
Seems good for me.

File: .github/PULL_REQUEST_TEPLATE.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+### Checklist
+
+<!-- Checklist for all PRs -->
+#### General
+- [ ] I chose a title conforming to the [naming conventions for pull requests](#).
+
+<!-- Checklist for PRs that affect our API-->
+#### AtlasML - API
+- [ ] I updated the API documentation.
```
Comment (ufukygmr) [inline .github/PULL_REQUEST_TEPLATE.md:27]: Another thing would be ``` I added the related tests```  for both API and ML part.

**Review by ufukygmr — CHANGES_REQUESTED** [review]
I thought 2 things regarding the templates.

File: .github/PULL_REQUEST_TEPLATE.md (hunk 1 above)
Comment (MaximilianAnzinger) [reply .github/PULL_REQUEST_TEPLATE.md:30]: I've already created documentation, see e.g.  #13 but I'm not finished with the deployment yet. Detailed links will follow as soon as a copy is deployed on our own instances.

**Pull request merged**